app/interactors/sync.rb

```diff
@@ -43,37 +68,6 @@ def results
 
   private
 
-  def find_existing_model(model)
-    lookup_args = []
-    lookup_hash = {}
-    matching_attributes.each do |attr|
-      if attr.instance_of?(Symbol)
-        lookup_value = model.send(attr)
-        lookup_hash[attr] = lookup_value if lookup_value
-      else
-        args = attr.map do |column, key|
-          lookup_attribute = model.send(column)
-          lookup_value = lookup_attribute[key]
-          h = Sequel.hstore_op(column)
-          h.contains(Sequel.hstore({key => lookup_value}))
-        end
-        lookup_args.concat(args)
-      end
-    end
-
-    lookup_args << lookup_hash unless lookup_hash.empty?
-
-    model_class.find(*lookup_args) unless lookup_args.empty?
-  end
-
-  def update_existing(existing_model, new_model)
-    values = new_model.values
-    filtered_values = values.reject { |key| skip_updating.include?(key) }
-    existing_model.update_all(filtered_values)
-    new_model.id ||= existing_model.id
-    existing_model
-  end
-
   def key_name
     self.class.key_name
```

Comment:
Aha, I overlooked it… and understand now. So what about some metaprogramming:

```rb
%i[model_class key_name matching_attributes skip_updating].each do |key|
  define_method(key) { self.class.send(key) }
end
```

_(I’m writing it from head, not sure if it’s gonna work.)_